transcom/mymove
# INT B-20949 SSW remaining ppm incentive calculation

## [B-20949](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A997044)

## Summary

**INTEGRATION**

This PR corrects the issue of the display value for the **_Remaining PPM Incentive (pre-tax)_** on page 2 of the SSW form for the Payment Download upon close out review. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as customer and create and submit a move containing 3 PPMs. This is to verify 3 test cases.
2. Login as service counselor and submit review details to start closeout process.
3. Login as customer click Download AOA Paperwork (PDF). Ensure all three AOA downloads have "N/A" in the **_Remaining PPM Incentive (pre-tax)_** section.
4. As customer start upload PPM documents for reimbursement process for each PPM.
5. First PPM set fictitious money you received when asked "How much did you receive?" Make a note of it. Then one or more Storage expenses and click GTCC payment was not used. Make note of numbers entered for storage expense. Submit.
6. Second PPM set fictitious money you received when asked "How much did you receive?" Make a note of it. Then add Storage expense and click GTCC payment was used. Submit.
7. Third PPM set fictitious money you received when asked "How much did you receive?" Make a note of it. Then add expense(s) NOT STORAGE and click GTCC payment was not used. Submit.
8. Login as SC and review documents for each PPM. Approve all.
9. Upon close out, each PPM will now have ability to download Payment Packet. Download each.
10. Verify display in **_Remaining PPM Incentive (pre-tax)_** in dollars. Note, this PR uses existing code/logic to compute this field.  It uses values currently display on the SSW form in production.  So for verification purposes use the following equation to verify if the display is "correct".
11. Make sure to go through permutations and uses you can think of that is applicable to this PR not specifically outlined in the testing steps.

X is "Remaining PPM Incentive (pre-tax)"
A is "Actual Obligations 100% GCC (lbs)"

B is "Member-paid SIT"
C is "GTCC-paid SIT"
D is "Advance"

X = (A + B + C) - D
 
Ex: 

![Screenshot 2024-08-28 at 1 32 25 PM](https://github.com/user-attachments/assets/d113b82f-96a7-4c6f-af27-1fbb83b3b429)



![Screenshot 2024-08-28 at 1 33 33 PM](https://github.com/user-attachments/assets/f5f1f0da-9a29-407e-a7d6-f32e8e2c3644)



![Screenshot 2024-08-28 at 1 34 59 PM](https://github.com/user-attachments/assets/a9b42edf-c26d-454c-8f40-9b0e4b8e903c)




![Screenshot 2024-08-28 at 1 35 59 PM](https://github.com/user-attachments/assets/2ec3d55e-d41f-469f-be04-68917f77ad9a)


`159.07 =109.07+75-25`

 


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
